src/main/java/ch/njol/skript/expressions/ExprYawPitch.java

```diff
@@ -35,118 +39,182 @@
+			return usesYaw
+					? toSkriptYaw(location.getYaw())
+					: toSkriptPitch(location.getPitch());
+		} else if (object instanceof Location) {
+			Location location = (Location) object;
+			return usesYaw
+					? toSkriptYaw(location.getYaw())
+					: toSkriptPitch(location.getPitch());
 		} else if (object instanceof Vector) {
-			Vector vector = ((Vector) object);
-			if (usesYaw)
-				return VectorMath.skriptYaw(VectorMath.getYaw(vector));
-			return VectorMath.skriptPitch(VectorMath.getPitch(vector));
+			Vector vector = (Vector) object;
+			return usesYaw
+					? toSkriptYaw(VectorMath.getYaw(vector))
+					: toSkriptPitch(VectorMath.getPitch(vector));
 		}
 		return null;
 	}
 
-	@SuppressWarnings({"null"})
 	@Override
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE)
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (getExpr().getReturnType().isAssignableFrom(Player.class) && !SUPPORTS_PLAYERS)
+			return null;
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
 			return CollectionUtils.array(Number.class);
+		} else if (mode == ChangeMode.RESET) {
+			return new Class[0];
+		}
 		return null;
 	}
 
-	@SuppressWarnings("null")
 	@Override
-	public void change(Event e, Object[] delta, ChangeMode mode) {
-		if (delta == null)
-			return;
+	public void change(Event event, Object[] delta, ChangeMode mode) {
 		float value = ((Number) delta[0]).floatValue();
-		for (Object single : getExpr().getArray(e)) {
-			if (single instanceof Location) {
-				changeLocation(((Location) single), value, mode);
-			} else if (single instanceof Vector) {
-				changeVector(((Vector) single), value, mode);
+		for (Object object : getExpr().getArray(event)) {
+			if (object instanceof Player && !SUPPORTS_PLAYERS)
+				continue;
+			if (object instanceof Entity) {
+				changeForEntity((Entity) object, value, mode);
+			} else if (object instanceof Location) {
+				changeForLocation(((Location) object), value, mode);
+			} else if (object instanceof Vector) {
+				changeForVector(((Vector) object), value, mode);
 			}
 		}
 	}
 
-	private void changeLocation(Location l, float value, ChangeMode mode) {
+	private void changeForEntity(Entity entity, float value, ChangeMode mode) {
+		Location location = entity.getLocation();
 		switch (mode) {
 			case SET:
-				if (usesYaw)
-					l.setYaw(convertToPositive(value));
-				else
-					l.setPitch(value);
+				if (usesYaw) {
+					entity.setRotation(fromSkriptYaw(value), location.getPitch());
+				} else {
+					entity.setRotation(location.getYaw(), fromSkriptPitch(value));
+				}
 				break;
+			case REMOVE:
+				value = -value;
 			case ADD:
-				if (usesYaw)
-					l.setYaw(convertToPositive(l.getYaw()) + value);
-				else
-					l.setPitch(l.getPitch() + value);
+				if (usesYaw) {
+					entity.setRotation(location.getYaw() + value, location.getPitch());
```

Comment:
Question: Shouldn't this use `+ fromSkriptYaw(value)` as well?

Reply:
No, we're adding a plain value and the values are not multiplied or scaled in any way.